### DISCUSSION TITLE
Is there a way to create table of all the measures and the dependents of each?

### DISCUSSION THREAD

**Discussion by Rangi96** [DaxStudio/DaxStudio#1089] [Q&A]
I am receiving someone else's Dashboard, and I want to follow the dependencies between measures easily. I would like a table with a column with all measures and a second column with measures that depend on the first.

**Comment by dgosbell** [comment]
You can get this information from the DISCOVER_CALC_DEPENDENCY DMV. The following query will give you all the dependencies, but you can filter the object type if you are just interested in measures.

`SELECT * FROM $SYSTEM.DISCOVER_CALC_DEPENDENCY`

**Comment by DavidUoS** [comment]
I just discovered this helpful table. To make my life a little easier I wanted to return only some of the columns. (This only saves me having to delete the ones I don't want from Excel, and if I'm using the results window makes that easier.)
This works fine:
select OBJECT_TYPE, OBJECT, EXPRESSION, REFERENCED_TABLE, REFERENCED_OBJECT, REFERENCED_EXPRESSION
from $SYSTEM.DISCOVER_CALC_DEPENDENCY
but this throws an error:
select OBJECT_TYPE, **TABLE,** OBJECT, EXPRESSION, REFERENCED_TABLE, REFERENCED_OBJECT, REFERENCED_EXPRESSION
from $SYSTEM.DISCOVER_CALC_DEPENDENCY
Presumably because TABLE is a keyword.
So how can I ask for a column called "TABLE" other than asking for them all?

**Reply by dgosbell** [reply]
@DavidUoS - you are correct TABLE is a keyword, you can fix this issue by wrapping it in square brackets to delimit it.
```
select OBJECT_TYPE, [TABLE], OBJECT, EXPRESSION, REFERENCED_TABLE, REFERENCED_OBJECT, REFERENCED_EXPRESSION
from $SYSTEM.DISCOVER_CALC_DEPENDENCY
```

**Reply by DavidUoS** [reply]
That is very helpful - thanks. I tried all sorts of delimiters and qualifications, but not []. Is there some documentation I can refer to that explains this fuzzy land between DAX and SQL?

**Reply by dgosbell** [reply]
The syntax used by the DMV queries in the tabular engine is a syntax called DMX. There is documentation for the DMX language [here](https://learn.microsoft.com/en-us/sql/dmx/data-mining-extensions-dmx-reference?view=sql-server-ver16) but the DMV queries only use a small portion of that language.

**Reply by DavidUoS** [reply]
Thanks again
